scripts/globals/fishing_contest.lua

```diff
@@ -0,0 +1,417 @@
+    [ 2] = { gil = 25000, item = xi.item.PELICAN_RING, title = xi.title.MYTHRIL_HOOK },
+    [ 3] = { gil = 10000, item = xi.item.PELICAN_RING, title = xi.title.SILVER_HOOK  },
+    [ 4] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 5] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 6] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 7] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 8] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [ 9] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [10] = { gil = 1000,  item = xi.item.PELICAN_RING, title = xi.title.COPPER_HOOK  },
+    [11] = {              item = xi.item.PELICAN_RING,                               },
+    [12] = {              item = xi.item.PELICAN_RING,                               },
+    [13] = {              item = xi.item.PELICAN_RING,                               },
+    [14] = {              item = xi.item.PELICAN_RING,                               },
+    [15] = {              item = xi.item.PELICAN_RING,                               },
+    [16] = {              item = xi.item.PELICAN_RING,                               },
+    [17] = {              item = xi.item.PELICAN_RING,                               },
+    [18] = {              item = xi.item.PELICAN_RING,                               },
+    [19] = {              item = xi.item.PELICAN_RING,                               },
+    [20] = {              item = xi.item.PELICAN_RING,                               },
+}
+
+xi.fishingcontest.interval =
+{
+    -- These intervals indicate how long each phase lasts before being automatically progressed.
+    -- Total cycle is 28 days, 35 minutes based on captures.  May be varied if contest is manually controlled
+    [xi.fishingcontest.status.CONTESTING] = 300,        -- 00 days 00 hours 05 mins
+    [xi.fishingcontest.status.OPENING]    = 1500,       -- 00 days 00 hours 25 mins
+    [xi.fishingcontest.status.ACCEPTING]  = 1209600,    -- 14 days 00 hours 00 mins
+    [xi.fishingcontest.status.RELEASING]  = 1800,       -- 00 days 00 hours 30 mins
+    [xi.fishingcontest.status.PRESENTING] = 1206000,    -- 13 days 23 hours 00 minutes
+    [xi.fishingcontest.status.HIATUS]     = 2100,       -- 00 days 00 hours 35 minutes
+    [xi.fishingcontest.status.CLOSED]     = 0xFFFFFFFF, -- Must be manually controlled
+}
+
+-----------------------------------
+-- LOCAL FUNCTIONS
+-----------------------------------
+
+-- Determine the slot that contains the fish being traded
+local function findFishSlot(trade, fish)
+    for i = 0, trade:getSlotCount() - 1 do
+        if trade:getItemId(i) == fish then
+            return i
+        end
+    end
+
+    return 0
+end
+
+-- The npcUtil.giveItem function currently does not support custom exdata
```

Comment:
Would be useful to add exdata to npcUtil.giveItem?

Reply:
I'm happy to do that if you want it in this PR, but yes -- it would definitely be useful (probably for others too, I assume?)